### PR TITLE
Add shared_array_buffer serialization

### PR DESCRIPTION
**Pull request by gergocs**
JerryScript-DCO-1.0-Signed-off-by: Gergo Csizi gergocs@inf.u-szeged.hu

### CODE REVIEW

File: deps/jerry/v8jerry/v8jerry_serialize.cpp [hunk 1]
```diff
@@ -231,8 +231,10 @@ bool ValueSerializer::WriteValueInternal(jerry_value_t value) {
         WriteJerryBigInt(value);
     } else if (jerry_value_is_string(value)) {
         WriteJerryString(value);
-    } else if (jerry_value_is_arraybuffer(value) || jerry_value_is_shared_arraybuffer(value)) {
+    } else if (jerry_value_is_arraybuffer(value)) {
         return WriteJerryArrayBuffer(value);
+    } else if(jerry_value_is_shared_arraybuffer(value)){
+        return GetJerrySharedArrayBufferId(value);
```
Comment (zherczeg) [inline deps/jerry/v8jerry/v8jerry_serialize.cpp]: GetJerrySharedArrayBufferId -> WriteJerrySharedArrayBuffer

File: deps/jerry/v8jerry/v8jerry_serialize.cpp [hunk 2]
```diff
@@ -627,15 +639,17 @@ jerry_value_t ValueDeserializer::ReadValueInternal() {
...
+        case SerializationTag::kArrayBuffer: {
+            jerry_value_t array_buffer = ReadJerryArrayBuffer();
             if (CheckTag(SerializationTag::kArrayBufferView)) {
                 ReadTag(&tag);
                 return ReadJerryArrayBufferView(array_buffer);
             }
             return array_buffer;
         }
+        case SerializationTag::kSharedArrayBuffer: {
+            return GetJerrySharedArrayBufferFromId();
```
Comment (zherczeg) [inline deps/jerry/v8jerry/v8jerry_serialize.cpp]: GetJerrySharedArrayBufferFromId -> ReadJerrySharedArrayBuffer

**Review by zherczeg — CHANGES_REQUESTED** [review]
Good patch, olny needs minor changes.

**Comment by gergocs** [comment]
Patch is ready for review

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**